### PR TITLE
[ACS-5839] migrate from QueryBody to SearchResult type

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- mirgate from QueryBody (type alias) to a real SearchResult type before QueryBody is removed (deprecated long time ago with JS-API 3.x)

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://alfresco.atlassian.net/browse/ACS-5839

### CODE REVIEW

File: lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts [hunk 1]
```diff
@@ -27,6 +26,5 @@ export interface SearchConfigurationInterface {
      * @param skipCount The offset of the start of the page within the results list
      * @returns Query body defined by the parameters
      */
-    generateQueryBody(searchTerm: string, maxResults: number, skipCount: number): QueryBody;
-
+    generateQueryBody(searchTerm: string, maxResults: number, skipCount: number): SearchRequest;
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts:29]: @DenysVuika there is search-configuration.interface.md file which still has QueryBody instead SearchRequest. Could you please correct that documentation file too to have it synchronized?

File: lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts [hunk 2]
```diff
@@ -15,10 +15,9 @@
  * limitations under the License.
  */
 
-import { QueryBody } from '@alfresco/js-api';
+import { SearchRequest } from '@alfresco/js-api';
 
 export interface SearchConfigurationInterface {
-
     /**
      * Generates a QueryBody object with custom search parameters.
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts:22]: Can you update this too?

File: lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts [hunk 3]
```diff
@@ -27,6 +26,5 @@ export interface SearchConfigurationInterface {
      * @param skipCount The offset of the start of the page within the results list
      * @returns Query body defined by the parameters
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts:27]: This probably should be updated too

File: lib/content-services/src/lib/search/services/search-configuration.service.ts [hunk 4]
```diff
@@ -32,8 +31,8 @@ export class SearchConfigurationService implements SearchConfigurationInterface
      * @param skipCount The offset of the start of the page within the results list
      * @returns Query body defined by the parameters
      */
-    generateQueryBody(searchTerm: string, maxResults: number, skipCount: number): QueryBody {
-        const defaultQueryBody: QueryBody = {
+    generateQueryBody(searchTerm: string, maxResults: number, skipCount: number): SearchRequest {
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/search/services/search-configuration.service.ts:34]: Could you please update search-configuration.service.md file with new types?

File: lib/content-services/src/lib/search/services/search.service.ts [hunk 5]
```diff
@@ -86,12 +87,14 @@ export class SearchService {
      * @param queryBody Object containing the search parameters
      * @returns List of search results
      */
-    searchByQueryBody(queryBody: QueryBody): Observable<ResultSetPaging> {
+    searchByQueryBody(queryBody: SearchRequest): Observable<ResultSetPaging> {
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/search/services/search.service.ts:90]: Could you please update search.service.md file with new types?

File: lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts (hunk 1 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts:29]: I can't change the public api yet as it's going to be a breaking change, while changing the return type seems to be relatively ok

File: lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts (hunk 3 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/common/interfaces/search-configuration.interface.ts:27]: it's just the concept here, not the type name

File: lib/content-services/src/lib/search/services/search-configuration.service.ts (hunk 4 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/search/services/search-configuration.service.ts:34]: updated docs, the public api stil remains the same for now

**Comment by DenysVuika** [comment]
@AleksanderSklorz updated the docs and variable names where possible

**Review by AleksanderSklorz — APPROVED** [review]

**Comment by AleksanderSklorz** [comment]
> @AleksanderSklorz updated the docs and variable names where possible

Thank you, approved.

**Review by nikita-web-ua — APPROVED** [review]

**Pull request merged**